### PR TITLE
JNG-4856 Add test cases for single and collection query

### PR DESCRIPTION
**Pull request by gaborflorian**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4856" title="JNG-4856" target="_blank">TaskJNG-4856</a>  Test static entity return type query dao functionality
  </td></table>
  <br />

### CODE REVIEW

**Review by noherczeg — APPROVED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java [hunk 1]
```diff
@@ -233,4 +274,235 @@ void testPrimitiveQueriesDao() {
...
+        //TODO JNG-4867 Count didn't work
+//        assertEquals(2, entityCollectionWithValueParamQueryDao
+//                .query(EntityCollectionWithValueParamQueryParameter
+//                        .builder()
+//                        .withValue(4)
+//                        .build())
+//                .count()
+//        );
+
+        ///With parameter default
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java]: /// ? :)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java [hunk 2]
```diff
@@ -233,4 +274,235 @@ void testPrimitiveQueriesDao() {
...
+                        .builder()
+                        .withValue(4)
+                        .build())
+                .orderBy(EntityQueryElementAttribute.NAME)
+                .execute();
+
+        assertEquals(d.identifier(), orderByList.get(0).identifier());
+        assertEquals(e.identifier(), orderByList.get(1).identifier());
+
+        //With parameter and orderByDesc
```
Comment (robertcsakany) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java]: Please put ' ' after '//'

**Review by robertcsakany — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java [hunk 3]
```diff
@@ -233,4 +274,235 @@ void testPrimitiveQueriesDao() {
...
+            "REQ-ENT-011"
+    })
+    void testEntitySingleQueriesDao() {
+
+        EntityQueryElement a = entityQueryElementDao.create(EntityQueryElement.builder().withName("A").withValue(1).withCategory(MyEnum.Bombastic).build());
+        EntityQueryElement b = entityQueryElementDao.create(EntityQueryElement.builder().withName("B").withValue(2).withCategory(MyEnum.Bombastic).build());
+        EntityQueryElement c = entityQueryElementDao.create(EntityQueryElement.builder().withName("C").withValue(3).withCategory(MyEnum.Crazy).build());
+        EntityQueryElement d = entityQueryElementDao.create(EntityQueryElement.builder().withName("D").withValue(4).withCategory(MyEnum.Atomic).build());
+
+        //TODO Should be not a list because it can be only one element.
```
Comment (bencegelei) [inline judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java]: Please include ticket number

**Review by bencegelei — CHANGES_REQUESTED** [review]

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java (hunk 1 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java]: [ddec3ca](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/176/commits/ddec3cad2dcf7d4f1aec862250edbc922d17a045)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java (hunk 2 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java]: [ddec3ca](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/176/commits/ddec3cad2dcf7d4f1aec862250edbc922d17a045)

File: judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java (hunk 3 above)
Comment (gaborflorian) [reply judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/TestStaticQueryDao.java]: [ddec3ca](https://github.com/BlackBeltTechnology/judo-runtime-core-jsl/pull/176/commits/ddec3cad2dcf7d4f1aec862250edbc922d17a045)

**Review by robertcsakany — APPROVED** [review]

**Review by bencegelei — APPROVED** [review]

**Review by Madmat8 — APPROVED** [review]

**Pull request merged**